docs/design/kademlia_audit_gating.md

```diff
@@ -0,0 +1,116 @@
+
+2. Create a trusted Satellite list that contains the IDs of Satellites from which Nodes will accept verification signatures of other Nodes.
+
+3. Implement a Routing Table Antechamber: An XOR-ordered data structure in which unverified nodes are entered if successfully contacted. 
+
+4. Modify FindNear to return n XOR-closest nodes from the antechamber in addition to those already returned from the routing table.
+
+5. On deployment, avoid complete erasure of the network's routing tables.
+
+
+## Terminology
+**Routing Table Antechamber** - *XOR-ordered temporary holding place for unverified storagenodes*
+
+**Node Neighborhood** - *The k-closest nodes to self where distance is measured by XOR. A node is within the node neighborhood if it is closer than the furthest node in the neighborhood. The vetted node neighborhood is the k-closest nodes that are currently in the Routing Table.*
+
+<img src="./kad-audit.jpg" alt="node neighborhood" width="500"/>
+
+## Design
+
+1. Satellite Signatures for Node Verification
+    - Identities can sign messages already
+    - Create a Voucher (message) that satellites can sign that a Node has been verified: qualification status, audit successes, and uptime
+    - Audit success ratio and uptime count thresholds are per-satellite
+    - The vouchers issued by the satellite should have an expiration on them (tunable by satellite)
+    - Nodes are expected to get up to date vouchers
+```go
+    // Satellite
+    // IsVetted returns whether or not the node reaches reputable thresholds
+    IsVetted(ctx context.Context, id storj.NodeID, criteria *NodeCriteria) (bool, error)
+    
+    // Storagenode
+    // DB implements storing and retrieving vouchers
+    type DB interface {
+    // Put inserts or updates a voucher from a satellite
+    Put(context.Context, *pb.Voucher) error
+    // GetExpiring retrieves all vouchers that are expired or about to expire
+    GetExpiring(context.Context) ([]storj.NodeID, error)
+    // GetValid returns one valid voucher from the list of approved satellites
+    GetValid(context.Context, []storj.NodeID) (*pb.Voucher, error)
+    }
+```
+
+2. Trusted Satellites List
+    - Create Whitelist/blacklist with an abstraction layer for trusted/untrusted Satellites
+    - These lists will live on each Node
+    - NB: We are using `config.Storage.WhitelistedSatelliteIDs` "a comma-separated list of approved satellite node ids" for now
+
+3. Routing Table Antechamber
+    - XOR ordered data structure (boltdb bucket)
+    - A node can be added if it would be within the vetted node neighborhood
+    - Once a node has been verified, it broadcasts its new status to try to join routing tables
+    - A Node may enter a Routing Table directly if at first contact it is already verified by a trusted Satellite.  
+    - A Node may be removed from a Routing Table if on bucket refresh it no longer meets qualifications
```

Comment:
nit: "may" is probably not the word you want here.  "should"?

Reply:
you're right